packages/gappa/gappa.1.4.1/opam

```diff
@@ -0,0 +1,45 @@
+opam-version: "2.0"
+maintainer: "7895506+MSoegtropIMC@users.noreply.github.com"
+authors: "Guillaume Melquiond"
+bug-reports: "https://gitlab.inria.fr/gappa/gappa/-/issues"
+homepage: "https://gitlab.inria.fr/gappa/gappa"
+dev-repo: "git+https://gitlab.inria.fr/gappa/gappa.git"
+license: "CECILL-2.1"
+patches: [
+  "0001-Added-configure-for-c-11.patch"
+]
+build: [
+  [ "autoreconf" ]
+  # Note: configure.in seems to reference this file
+  [ "touch" "stamp-config_h.in" ]
+  [ "./configure"
+    # If someone knows how to ask MacPorts for the usual include and lib path, please tell me
+    "CXXFLAGS=-I/opt/local/include" { os-distribution = "macports" & os = "macos" }
+    "LDFLAGS=-L/opt/local/lib" { os-distribution = "macports" & os = "macos" }
+    # Support installing on Apple Silicon with Homebrew
+    "CXXFLAGS=-I/opt/homebrew/include" { os-distribution = "homebrew" & os = "macos" & arch = "arm64"}
+    "LDFLAGS=-L/opt/homebrew/lib" { os-distribution = "homebrew" & os = "macos" & arch = "arm64"}
+    # Support for windows
+    "--build=%{arch}%-pc-cygwin" { os = "win32" & os-distribution = "cygwinports" }
+    "--host=%{arch}%-w64-mingw32" { os = "win32" & os-distribution = "cygwinports" }
+    "--target=%{arch}%-w64-mingw32" { os = "win32" & os-distribution = "cygwinports" }
+    "--prefix=%{prefix}%" 
+  ]
+  [ "./remake" "--jobs=%{jobs}%" ]
```

Comment:
This custom build system doesn’t work on Alpine (tmpnam doesn’t take the `TMPDIR` env variable into account and thus fail with the sandbox) and won’t work on macOS either given any network (even local) access is forbidden by the macOS sandbox (builtin macOS sandbox restriction).

Is there any chance you could be using something a little less custom to build your package?

Reply:
It definitely does work on MacOS with MacPorts and Homebrew - I am using both myself.

With "custom build system" you refer to `remake` - it is not that uncommon. If there are issues with it, I guess I can have them fixed.

Reply:
I can look into what issues conf-bison and conf-flex have on Homebrew MacOS, but it seems to be a CI artefact. It works fine in GitHub based MacOS CI and it also works fine on my private machine (I have a x86 Homebrew and a ARM Macports machine). I will check on my private Homebrew machine if the installation (not just checking) of bison and flex works.

Reply:
@kit-ty-kate : I had a look at Alpine. As far as I understand it uses this implementation for `tempname`:

https://git.musl-libc.org/cgit/musl/tree/src/stdio/tempnam.c

which indeed does not check any environment variables. Doesn't this lead to many other issues with opam?

I can ask the author of remake if they want to work around this. Otherwise I would state that a light wight OS like Alpine is not the target audience for proof generators.

Reply:
> It definitely does work on MacOS with MacPorts and Homebrew - I am using both myself.
> 
> With "custom build system" you refer to `remake` - it is not that uncommon. If there are issues with it, I guess I can have them fixed.

via opam? I'm surprised

Reply:
Yes, via opam. Coq Platform is completely opam based and I have gappa in Coq Platform CI since years, which tests on MacOS (homebrew), Ubuntu and Windows. There was never any problem with this package.

Do you have a pointer to the documentation of what should be writeable?